microdnf/context.hpp

```diff
@@ -103,6 +109,10 @@ void download_packages(libdnf::base::Transaction & transaction, const char * des
 
 void run_transaction(libdnf::rpm::Transaction & transaction);
 
+/// Parses the `spec` and adds it to the appropriate set. Duplicates are ignored.
+/// Returns `true` if the `spec` was inserted into a set.
+bool parse_add_spec(const std::string & spec, std::set<std::string> & pkg_specs, std::set<std::string> & filepaths);
```

Comment:
I am not sure but keeps c++ set order of insertion of elements? In python id does not keep the order.

Reply:
No. Items inserted into std::set sorted.

Reply:
OK. After discussion, we concluded that the order is not irrelevant. In rare cases, libsolv provides different results for different orders. I reworked the code.

Reply:
Could you document the rare cases in a comment?